mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1722,25 +1723,22 @@ private class LogisticAggregator(
     var maxMargin = Double.NegativeInfinity
 
     val margins = new Array[Double](numClasses)
+    val featureStdArray = new Array[Double](features.size)
     features.foreachActive { (index, value) =>
-      val stdValue = value / localFeaturesStd(index)
-      var j = 0
-      while (j < numClasses) {
-        margins(j) += localCoefficients(index * numClasses + j) * stdValue
-        j += 1
-      }
+      featureStdArray(index) = value / localFeaturesStd(index)
```

Comment:
Here why don't handle the case `localFeaturesStd(index) == 0.0` ?
I remember other place it handle such case.